Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -51,19 +81,290 @@ namespace AZ::SceneAPI::Behaviors
+
+        for (const auto& entry : meshTransformMap)
+        {
+            const auto thisNodeIndex = entry.first;
+            const auto meshNodeIndex = entry.second.first;
+            const auto meshNodeName = graph.GetNodeName(meshNodeIndex);
+            AZStd::string meshNodePath { meshNodeName.GetPath() };
+
+            AZStd::string meshNodeFullName;
+            meshNodeFullName = relativeSourcePath;
+            meshNodeFullName.append("_");
+            meshNodeFullName.append(meshNodeName.GetName());
+
+            auto meshGroup = AZStd::make_shared<AZ::SceneAPI::SceneData::MeshGroup>();
+            meshGroup->SetName(meshNodeFullName.c_str());
+            meshGroup->GetSceneNodeSelectionList().AddSelectedNode(AZStd::move(meshNodePath));
+            for (const auto& meshGoupNamePair : meshTransformMap)
+            {
+                if (meshGoupNamePair.first != thisNodeIndex)
+                {
+                    const auto nodeName = graph.GetNodeName(meshGoupNamePair.second.first);
+                    meshGroup->GetSceneNodeSelectionList().RemoveSelectedNode(nodeName.GetPath());
+                }
+            }
+            meshGroup->OverrideId(DataTypes::Utilities::CreateStableUuid(
+                scene,
+                azrtti_typeid<AZ::SceneAPI::SceneData::MeshGroup>(),
+                meshNodeName.GetPath()));
+
+            // this clears out the mesh coordinates each mesh group will be rotated and translated
+            // using the attached scene graph node
+            auto coordinateSystemRule = AZStd::make_shared<AZ::SceneAPI::SceneData::CoordinateSystemRule>();
+            coordinateSystemRule->SetUseAdvancedData(true);
+            coordinateSystemRule->SetRotation(AZ::Quaternion::CreateIdentity());
+            coordinateSystemRule->SetTranslation(AZ::Vector3::CreateZero());
+            coordinateSystemRule->SetScale(1.0f);
+            meshGroup->GetRuleContainer().AddRule(coordinateSystemRule);
+
+            manifestUpdates.emplace_back(meshGroup);
+
+            // create an entity for each MeshGroup
+            AZ::EntityId entityId;
+            AzToolsFramework::EntityUtilityBus::BroadcastResult(
+                entityId,
+                &AzToolsFramework::EntityUtilityBus::Events::CreateEditorReadyEntity,
+                meshNodeName.GetName());
+
+            if (entityId.IsValid() == false)
+            {
+                return Events::ProcessingResult::Ignored;
+            }
+
+            // Since the mesh component lives in a gem, then create it by name
+            AzFramework::BehaviorComponentId editorMeshComponent;
+            AzToolsFramework::EntityUtilityBus::BroadcastResult(
+                editorMeshComponent,
+                &AzToolsFramework::EntityUtilityBus::Events::GetOrAddComponentByTypeName,
+                entityId,
+                "{DCE68F6E-2E16-4CB4-A834-B6C2F900A7E9} AZ::Render::EditorMeshComponent");
```

Comment:
is the gem this component lives in in the dependency list for the prefab gem? I don't think I see it, it should be added if it's not https://github.com/o3de/o3de/blob/development/Gems/Prefab/PrefabBuilder/CMakeLists.txt

Reply:
There is no hard dependency on the Atom gems since they are not really required. The editorMeshComponent will be invalid so the mesh component will be skipped.

I will add a block to skip the default construction if that component can not be found.